_plugins/api.rb

```diff
@@ -85,6 +93,38 @@ def self.generate_locations_endpoint(site)
         'Index of team members by location code', data)
     end
 
+    # Based off of
+    # https://github.com/jekyll/jekyll/blob/v2.5.2/test/test_page.rb#L10-L13
+    # Modifies `page`
+    def self.do_render(site, page)
+      layout = Jekyll::Layout.new(site, '_layouts', 'simple.html')
+      layouts = Hash.new(layout)
+      page.render(layouts, site.site_payload)
+    end
+
+    def self.pages(site)
+      site.pages.select {|page| %w(.html .md).include?(page.ext) }
+    end
+
+    def self.get_output(site, page)
+      do_render(site, page)
+      filterer = Filterer.new
+      filterer.condense(filterer.strip_html(page.content))
+    end
```

Comment:
@parkr @mattr- I know you don't have a lot of context here, but in short: how do I access the fully rendered content of a `Page` within a plugin? The code above is close, though the liquid tags are still present. Any ideas? I looked through some existing Jekyll plugins, but didn't see anything that attempted this.

Side note: even with prodding around the Jekyll source code, getting this far was difficult; the `Page`s having an implicit state of being `#render`ed or not is pretty confusing. Going to try and take what I learn(ed) here and do a PR on Jekyll to make the `content`/`output`/`render` methods idempotent. Just FYI :wink: